### PR TITLE
fix: set `Twitch-Eventsub-Subscription-Type` to correct type

### LINKED ISSUE #163: EventSub trigger sets `Twitch-Eventsub-Subscription-Type` to empty string
Running `twitch event trigger channel.channel_points_custom_reward_redemption.add -F http://localhost:8080/eventsub -s {secret}`, will set the `Twitch-Eventsub-Subscription-Type` header to `''`.

Running the command will execute with the following values:

`args[0] = 'channel.channel_points_custom_reward_redemption.add'`
https://github.com/twitchdev/twitch-cli/blob/f149736ea3ae7be26942d4c1fc60377d7e44fb83/cmd/events.go#L141
`p.Event = 'channel.channel_points_custom_reward_redemption.add', p.Transport = 'eventsub'`
https://github.com/twitchdev/twitch-cli/blob/f149736ea3ae7be26942d4c1fc60377d7e44fb83/internal/events/trigger/trigger_event.go#L79
`e = channel_points_redemption::Event`
https://github.com/twitchdev/twitch-cli/blob/f149736ea3ae7be26942d4c1fc60377d7e44fb83/internal/events/trigger/trigger_event.go#L121
`transport = 'eventsub', trigger = 'channel.channel_points_custom_reward_redemption.add'` ⚠️ this isn't a key into the map
https://github.com/twitchdev/twitch-cli/blob/f149736ea3ae7be26942d4c1fc60377d7e44fb83/internal/events/types/_template/_event_name.go#L63-L65
https://github.com/twitchdev/twitch-cli/blob/f149736ea3ae7be26942d4c1fc60377d7e44fb83/internal/events/types/channel_points_redemption/redemption_event.go#L20-L25
`p.Event = ''`
https://github.com/twitchdev/twitch-cli/blob/f149736ea3ae7be26942d4c1fc60377d7e44fb83/internal/events/trigger/forward_event.go#L71

### PR DESCRIPTION
**Pull request by Nerixyz**
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

## Problem/Feature

Fixes #163.

## Description of Changes: 

- Remove the call to `GetTopic` since at that point, we alredy have the event type.

## Checklist

- [x] My code follows the [Contribution Guide](https://github.com/twitchdev/twitch-cli/blob/master/CONTRIBUTING.md#Requirements)
- [x] I have self-reviewed the changes being requested
- [x] I have made comments on pieces of code that may be difficult to understand for other editors
- [x] I have updated the documentation (if applicable)

### CODE REVIEW

File: internal/events/trigger/trigger_event.go [hunk 1]
```diff
@@ -118,7 +118,7 @@ func Fire(p TriggerParameters) (string, error) {
 			JSON:           resp.JSON,
 			Secret:         p.Secret,
 			ForwardAddress: p.ForwardAddress,
-			Event:          e.GetTopic(p.Transport, p.Event),
+			Event:          p.Event,
```
Comment (lleadbet) [inline internal/events/trigger/trigger_event.go:121]: Removing this has a byproduct; the CLI also supports shorthand aliases for most of the events, such as `drop` or `add-redemption`. 

By bypassing that here, while it fixes your specific error, it breaks those users as it uses this to get the Topic. 

The fix here is to add a safety check here to check if `GetTopic` is an empty string (invalid shorthand) and then get the `GetEventSubAlias`  to see if it returns a value (which returns the shorthand). Basically: 

```go
		topic := e.GetTopic(p.Transport, p.Event)
		if topic == "" && e.GetEventSubAlias(p.Event) != "" {
			topic = p.Event 
		}
```

**Review by lleadbet — CHANGES_REQUESTED** [review]

File: internal/events/trigger/trigger_event.go (hunk 1 above)
Comment (Nerixyz) [reply internal/events/trigger/trigger_event.go]: I didn't know about shorthands 😄 fixed it.

**Comment by Nerixyz** [comment]
I just found that `retrigger` didn't always use the correct event (it sometimes errored with `Error refiring event: Invalid event`).

I think `retrigger` should print the response body as well, though this should be done in another PR.

**Comment by lleadbet** [comment]
> I just found that `retrigger` didn't always use the correct event (it sometimes errored with `Error refiring event: Invalid event`).
> 
> I think `retrigger` should print the response body as well, though this should be done in another PR.

Can you elaborate on this? Not sure if this was broken at some point in the past or with this PR.

**Review by lleadbet — APPROVED** [review]
Looks good, but need more info on the issue reported before merging.

**Comment by Nerixyz** [comment]
> Can you elaborate on this? Not sure if this was broken at some point in the past or with this PR.

I did
```
twitch event trigger channel.channel_points_custom_reward_redemption.add -F http://localhost:8080/eventsub -s <secret>
```
then with the event-id:
```
twitch event retrigger -F http://localhost:8080/eventsub -s <secret> -i <event-id>
```
which resulted in `Error refiring event: Invalid event` (maybe I'm doing something wrong though).

File: internal/events/trigger/retrigger_event.go [hunk 2]
```diff
@@ -21,19 +21,24 @@ func RefireEvent(id string, p TriggerParameters) (string, error) {
 
 	p.Transport = res.Transport
 
-	e, err := types.GetByTriggerAndTransport(p.Event, p.Transport)
+	e, err := types.GetByTriggerAndTransport(res.Event, p.Transport)
```
Comment (lleadbet) [inline internal/events/trigger/retrigger_event.go:24]: This actually fixes the bug you noted; `p.Event` doesn't have the trigger, only the database `res`ponse. 

Will merge this.

**Pull request merged**